### PR TITLE
Various small CI improvements

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6554)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: test/k8sT/manifests/cilium-ds-patch-geneve.yaml [hunk 1]
```diff
@@ -19,5 +19,4 @@ spec:
       volumes:
       - name: etcd-secrets
         secret:
-          secretName: cilium-etcd-client-tls
-      dnsPolicy: ClusterFirstWithHostNet
+          optional: false
```
Comment (aanm) [inline test/k8sT/manifests/cilium-ds-patch-geneve.yaml]: I think you have deleted more lines than you want it to.

**Comment by aanm** [comment]
test-me-please

```
00:47:24.472       Tests NodePort with L7 Policy [It]
00:47:24.472       /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
00:47:24.472 
00:47:24.472       Pod "testclient-jbtt8" can not connect to service "http://10.103.26.105:10080"
00:47:24.472       Expected command: kubectl exec -n default testclient-jbtt8 -- curl -s -D /dev/stderr --fail --connect-timeout 3 --max-time 8 http://10.103.26.105:10080 -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
00:47:24.472       To succeed, but it failed:
00:47:24.472       Exitcode: 28 
00:47:24.472       Stdout:
00:47:24.472        	 time-> DNS: '0.000020()', Connect: '0.000000',Transfer '0.000000', total '3.002144'
00:47:24.472       Stderr:
00:47:24.472        	 command terminated with exit code 28
00:47:24.472       	 
```

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
CI is failing because `WaitforNPods` is looking for an exact match not a minimal number. Will fix.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**